pkg/util/tls-crypto-watch/tls-crypto-watch.go

```diff
@@ -0,0 +1,192 @@
+	informers "kubevirt.io/containerized-data-importer/pkg/client/informers/externalversions"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+)
+
+// CryptoConfig contains TLS crypto configurables
+type CryptoConfig struct {
+	CipherSuites []uint16
+	MinVersion   uint16
+}
+
+// CdiConfigTLSWatcher is the interface of cdiConfigTLSWatcher
+type CdiConfigTLSWatcher interface {
+	GetCdiTLSConfig() *CryptoConfig
+	GetInformer() cache.SharedIndexInformer
+}
+
+type cdiConfigTLSWatcher struct {
+	// keep this around for tests
+	informer cache.SharedIndexInformer
+
+	config *CryptoConfig
+	mutex  sync.RWMutex
+}
+
+// NewCdiConfigTLSWatcher crates a new cdiConfigTLSWatcher
+func NewCdiConfigTLSWatcher(ctx context.Context, cdiClient cdiclient.Interface) CdiConfigTLSWatcher {
+	cdiInformerFactory := informers.NewFilteredSharedInformerFactory(cdiClient,
+		common.DefaultResyncPeriod,
+		metav1.NamespaceAll,
+		func(options *metav1.ListOptions) {
+			options.FieldSelector = "metadata.name=" + common.ConfigName
+		},
+	)
+
+	cdiConfigInformer := cdiInformerFactory.Cdi().V1beta1().CDIConfigs().Informer()
+
+	ctw := &cdiConfigTLSWatcher{
+		informer: cdiConfigInformer,
+		config:   DefaultCryptoConfig(),
+	}
+
+	cdiConfigInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			klog.V(3).Infof("cdiConfigInformer add callback: %+v", obj)
+			ctw.updateConfig(obj.(*cdiv1.CDIConfig))
+		},
+		UpdateFunc: func(_, obj interface{}) {
+			klog.V(3).Infof("cdiConfigInformer update callback: %+v", obj)
+			ctw.updateConfig(obj.(*cdiv1.CDIConfig))
+		},
+		DeleteFunc: func(obj interface{}) {
+			config := obj.(*cdiv1.CDIConfig)
+			klog.Errorf("CDIConfig %s deleted", config.Name)
+		},
+	})
+
+	go cdiInformerFactory.Start(ctx.Done())
+
+	klog.V(3).Infoln("Waiting for cache sync")
+	cache.WaitForCacheSync(ctx.Done(), cdiConfigInformer.HasSynced)
+	klog.V(3).Infoln("Cache sync complete")
+
+	return ctw
+}
+
+func (ctw *cdiConfigTLSWatcher) GetCdiTLSConfig() *CryptoConfig {
+	ctw.mutex.RLock()
+	defer ctw.mutex.RUnlock()
+	return ctw.config
+}
+
+func (ctw *cdiConfigTLSWatcher) GetInformer() cache.SharedIndexInformer {
+	return ctw.informer
+}
+
+func (ctw *cdiConfigTLSWatcher) updateConfig(config *cdiv1.CDIConfig) {
+	newConfig := &CryptoConfig{}
+
+	cipherNames, minTypedTLSVersion := SelectCipherSuitesAndMinTLSVersion(config.Spec.TLSSecurityProfile)
+	minTLSVersion, _ := ocpcrypto.TLSVersion(string(minTypedTLSVersion))
+	ciphers := CipherSuitesIDs(cipherNames)
+	newConfig.CipherSuites = ciphers
+	newConfig.MinVersion = minTLSVersion
+
+	ctw.mutex.Lock()
+	defer ctw.mutex.Unlock()
+	ctw.config = newConfig
+}
+
+// SelectCipherSuitesAndMinTLSVersion returns cipher names and minimal TLS version according to the input profile
+func SelectCipherSuitesAndMinTLSVersion(profile *ocpconfigv1.TLSSecurityProfile) ([]string, ocpconfigv1.TLSProtocolVersion) {
```

Comment:
My big question with this PR, and it's really probably not a big deal, is would it be better to return some defaults here, or defer to whatever the default golang is using.  Worst case theoretically, a cipher is compromised and we don't know about it.  Go library would stop using the cipher when we update the compiler.

Reply:
So these defaults are based on https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28recommended.29
Which is advised as the "recommended" for the vast majority of services.

But I also wouldn't mind omitting and deferring to go. Just want to make sure that is what we want